include/utilities.h

```diff
@@ -223,3 +224,48 @@ inline wide_argument_string widen_argument(const wchar_t* str) noexcept
 {
     return wide_argument_string{ str };
 }
+
+template <typename CharT>
+errno_t strcatImpl(CharT* dest, rsize_t destBufSize, CharT const* src)
+{
+    if (std::is_same<CharT, char>::value)
+    {
+        return strcat_s((char*)dest, destBufSize, (const char*)src);

```

Comment:
Do not use C-Style casts.

Reply:
Here c-style casting is done only after confirming the casting is safer to be done. CharT was already confirmed to be of type char* and then casting is done. So, I thought c++ style casting would not give an advantage over c-style in this case.

Reply:
I see and understand you stance.  However, C++ style casting has the benefit of telling the reader what kind of cast is happening.  Also, where is the check to see if the cast is safe?  If I can't find it, other people can't either.

Reply:
Agree, I will change it to c++ style casting. 
By confirming the casting is safer, i meant template parameter of the parameters used in strcat_s(CharT) is checked before casting.